pkg/client/src/app/queries/stakeholdergoups.ts

```diff
@@ -0,0 +1,61 @@
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { Stakeholder, StakeholderGroup } from "@app/api/models";
+import {
+  deleteStakeholder,
+  deleteStakeholderGroup,
+  getStakeholderGroups,
+} from "@app/api/rest";
+import { AxiosError } from "axios";
+
+export interface IStakeholderGroupFetchState {
+  stakeholderGroups: StakeholderGroup[];
+  isFetching: boolean;
+  fetchError: any;
+  refetch: any;
+}
+export interface IStakeholderGroupMutateState {
+  mutate: any;
+  isLoading: boolean;
+  error: any;
+}
+
+export const StakeholderGroupsQueryKey = "stakeholderGroups";
+
+export const useFetchStakeholderGroups = (): IStakeholderGroupFetchState => {
+  const { data, isLoading, error, refetch } = useQuery(
+    StakeholderGroupsQueryKey,
+    async () => (await getStakeholderGroups()).data,
+    {
+      onError: (error) => console.log("error, ", error),
+    }
+  );
+  return {
+    stakeholderGroups: data || [],
+    isFetching: isLoading,
+    fetchError: error,
+    refetch,
+  };
+};
+
+export const useDeleteStakeholderGroupMutation = (
+  onSuccess: (res: any) => void,
+  onError: (err: AxiosError) => void
```

Comment:
No need to propagate AxiosError this far. The `deleteStakeholderGroup` could return a Promise<AxiosResponse<StakeholderGroup>> which would propagate a standard Error.
Check out task queries for examples.

Reply:
If we know the error returned is an Axios Error with the deleteStakeholder call being an axios call, what would be the purpose of a generic error message being returned?